### PR TITLE
Refaсtor ActivationLayerTest

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refaсtor ActivationLayerTest
 - Update cpu plugin instantiation of these tests to use new implementation
 
### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/normalize.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ class NormalizeL2LayerCPUTest : public testing::WithParamInterface<NormalizeL2La
             ov::Tensor tensor;
             if (funcInput.get_element_type().is_real()) {
                 tensor = ov::test::utils::create_and_fill_tensor(
-                        funcInput.get_element_type(), targetInputStaticShapes[i], 10, -5, 7, 222);
+                        funcInput.get_element_type(), targetInputStaticShapes[i], 5, 0, 7, 222);
```
Comment (olpipi) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/normalize.cpp:93]: After changes in fill_data_random, this test start to fail. So have to fix it as well

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 2]
```diff
@@ -200,6 +200,8 @@ std::vector<std::string> disabledTestPatterns() {
         // Issue: 121313
         R"(smoke_GroupConvBackpropData.*paddingDefined/GroupConvBackpropLayerTest.Inference.*f16.*)",
         R"(smoke_GroupConvBackpropData.*paddingDefined/GroupConvBackpropLayerTest.Inference.*f32.*)",
+
```
Comment (iefode) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:203]: Ticket number?

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 3]
```diff
@@ -83,60 +83,65 @@ ov::runtime::Tensor generate(const ov::element::Type& elemType,
...
+ov::runtime::Tensor generate(const std::shared_ptr<ngraph::op::v0::HardSigmoid>& node,
                              size_t port,
                              const ov::element::Type& elemType,
                              const ov::Shape& targetShape) {
-    return Activation::generate(elemType, targetShape, InputGenerateData(-1000, 2000, 32768, 1));
-}
+    switch (port) {
+        case 1: {
+            std::vector<float> alpha(node->get_input_shape(1).size(), 0.2f);
+            return ov::test::utils::create_tensor<float>(elemType, targetShape, alpha, alpha.size());
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:93]: Hm. Lets generate tensor via `fill_tensor` method? The method is able to fill tensor by same values

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 4]
```diff
@@ -83,60 +83,65 @@ ov::runtime::Tensor generate(const ov::element::Type& elemType,
...
+        }
+        case 2: {
+            std::vector<float> beta(node->get_input_shape(2).size(), 0.5f);
+            return ov::test::utils::create_tensor<float>(elemType, targetShape, beta, beta.size());
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:97]: the same comment

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp [hunk 5]
```diff
@@ -223,7 +223,7 @@ void inline fill_data_random(T* pointer,
     const uint32_t k_range = k * range;  // range with respect to k
     random.Generate(k_range);
 
-    if (start_from < 0 && !std::is_signed<T>::value) {
+    if (start_from < 0 && std::is_arithmetic<T>::value && !std::is_signed<T>::value) {
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:226]: Why do you add `is_arithmetic` check?

Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:226]: **std::is_signed<T>::value == false** for **ov::float16** and **ov::bfloat16**.
So for these 2 types, we don't need to reset start_from to 0

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 3 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:93]: I cannot find this method. Could you please show what header contains it?

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:97]: I cannot find this method. Could you please show what header contains it?

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp (hunk 2 above)
Comment (olpipi) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp]: done

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 3 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp]: done

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp]: done

**Comment by vurusovs** [comment]
`ie_tests_cpu_macos_arm64_release` has passed

**Review by vurusovs — APPROVED** [review]

**Pull request merged**